packages/block-editor/src/components/border-radius-control/linked-button.js

```diff
@@ -6,20 +6,20 @@ import { link, linkOff } from '@wordpress/icons';
 import { __ } from '@wordpress/i18n';
 
 export default function LinkedButton( { isLinked, ...props } ) {
-	const label = isLinked ? __( 'Unlink Radii' ) : __( 'Link Radii' );
+	const label = isLinked ? __( 'Unlink radii' ) : __( 'Link radii' );
 
 	return (
 		<Tooltip text={ label }>
-			<Button
-				{ ...props }
-				className="component-border-radius-control__linked-button"
-				isPrimary={ isLinked }
-				isSecondary={ ! isLinked }
-				isSmall
-				icon={ isLinked ? link : linkOff }
-				iconSize={ 16 }
-				aria-label={ label }
-			/>
+			<span>
```

Comment:
The extra span, like it does for the other instances of this button, makes the button "not have text", which invokes the `:not(.has-text)` CSS.

Reply:
I'm also unclear as to why this `span` wrapper is required. It looks like it's been copied from the approach in the `BoxControl` or because the new spacing presets UI needed one? Removing the wrapper didn't appear to make any visual change to me.

| With `span` | Without `span` |
|---|---|
| <img width="275" alt="Screen Shot 2022-09-15 at 4 43 04 pm" src="https://user-images.githubusercontent.com/60436221/190336237-6df03d4b-8b4f-448b-8ecf-d2018ecf7845.png"> | <img width="272" alt="Screen Shot 2022-09-15 at 4 43 42 pm" src="https://user-images.githubusercontent.com/60436221/190336253-ef9ba9f1-2c48-4b87-836f-b552cdc5db5a.png"> |

If it isn't required, I'd vote to clean that up as well. Seeing if the other components (spacing presets and box control) still need the spans wrapping their buttons could be a follow-up though.

Reply:
Inded that's curious. It was definitely needed at one point, but it doesn't appear to be anymore. Removed!

Reply:
Actually my mistake, I removed it only from one of them. When I removed it from the border version, this happened:

![without span](https://user-images.githubusercontent.com/1204802/190387740-99cc4ba5-82e0-4e84-b1a7-5bd65a9ca6b6.gif)

Let me look at a different solution than adding a span.

Reply:
Oh, it's because for some reason, just the border-radius-link button gets a `has-text` CSS class applied, even though it shouldn't. 🤔

Reply:
Yeah I actually don't know how to fix this. When the span is present, the `has-text` class is not. When the span is removed, `has-text` appears. And that means the following icon-button-CSS doesn't apply anymore:

```
	.components-button.is-small.has-icon:not(.has-text) {
		min-width: $icon-size;
		padding: 0;
	}
```

Reply:
@ciampo if you have any insights, I'd appreciate it. This appears to be the button piece that adds the problematic class:

```
		'has-text': !! icon && !! children,
```

Feel free to push to the branch in case you have a fix.

Reply:
I was just looking into this.
Curiously, if the `Button` component is directly placed under the `ToolTip`, it has `null` children.

![check](https://user-images.githubusercontent.com/54422211/190401617-0bbef8a0-d22d-4a0d-b06e-38484a336ce6.png)

This would cause it to be considered as having text.
Inserting a span tag would solve this problem, but I'll look into it some more.

Reply:
Thank you 🙏

Reply:
@cianpo

I believe this is probably `Tooltip` merging refs.
This may be a problem that should be resolved on the `Tooltip` component, but I have tried an approach to check if the child is empty on the `Button` component.

This change resolves the issue, but is this appropriate?

```diff
--- a/packages/components/src/button/index.js
+++ b/packages/components/src/button/index.js
@@ -92,6 +92,8 @@ export function Button( props, ref ) {
                'components-button__description'
        );
 
+       const hasChildren = children?.[ 0 ] && children[ 0 ] !== null;
+
        const classes = classnames( 'components-button', className, {
                'is-secondary': variant === 'secondary',
                'is-primary': variant === 'primary',
@@ -101,7 +103,7 @@ export function Button( props, ref ) {
                'is-busy': isBusy,
                'is-link': variant === 'link',
                'is-destructive': isDestructive,
-               'has-text': !! icon && !! children,
+               'has-text': !! icon && hasChildren,
                'has-icon': !! icon,
        } );
```

Reply:
> @ciampo if you have any insights, I'd appreciate it. This appears to be the button piece that adds the problematic class:

I don't think I've ever really worked on `Button` since I started working on components, so I'd need to look into it specifically to understand what's going on.

Although the fix proposed above seems reasonable! @t-hamano, would you be able to open a PR with your suggested code changes and ping me there, so that we can assess them and hopefully merge the fix?

Reply:
Should we just include it in this PR? Otherwise can we land this one _with_ the span intact (PR as-is now), and then remove the span in that new separate PR?

Reply:
I'm asking because it's pretty crucial we get this PR merged before the freeze, so anything we can do to land this one sooner rather than later would be great. The "has-text" issue seems like a bug that can be fixed after the freeze as well.

Reply:
I will begin work on a PR for the button.
If @jasmussen's PR should be merged first, I think there is no problem with reinstating the `span` tag.
I think it's not too late to decide whether to remove `span` tag or not after the two PRs have been merged in the end.

Reply:
I have submitted #44198, but I need to do further validation because some tests are failing 😅
I vote to proceed with this well-reviewed PR merge.